service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -507,4 +519,26 @@ public DynamicList buildDynamicHearingList() {
     public DynamicList buildDynamicHearingList(UUID selected) {
         return asDynamicList(getHearingDetails(), selected, hearingBooking -> hearingBooking.toLabel(DATE));
     }
+
+    private final HearingType hearingType;
```

Comment:
Maybe not now but I noticed that there was a tech debt for clean up. Thoughts on combining all these new fields into one model class and then annotating that class in here with `@JsonUnwrapped`. This means that then all the fields are cleaned up in one class from our perspective, and from the perspective of CCD they are still all top level data fields.
e.g.
```java
public class ManageHearingBuilder {
    private final HearingType hearingType;
    private final String hearingTypeDetails;
    private final String hearingVenue;
    private final Address hearingVenueCustom;
    private final String firstHearingFlag; //also used for logic surrounding legacy hearings
    private final PreviousHearingVenue previousHearingVenue;
    private String previousVenueId;
    private final String noticeOfHearingNotes;
    private final Object hearingDateList;
    private final String hasExistingHearings;

    @TimeNotMidnight(message = "Enter a valid start time", groups = HearingDatesGroup.class)
    @Future(message = "Enter a start date in the future", groups = HearingDatesGroup.class)
    private final LocalDateTime hearingStartDate;

    @TimeNotMidnight(message = "Enter a valid end time", groups = HearingDatesGroup.class)
    @Future(message = "Enter an end date in the future", groups = HearingDatesGroup.class)
    private final LocalDateTime hearingEndDate;
    private final String sendNoticeOfHearing;
    private final HearingOptions hearingOption;
    private final List<ProceedingType> proceedingType;
}
```
then in case data it would be something like
```java
    private ManageHearingBuilder manageHearingBuilder;
```

Reply:
This object could also contain the set of keys to be removed as a static field.

Reply:
This could also replace the use of a map to populate initial fields, then in the controller just convert this object to a map and add that map to data.
```java
ManageHearingBuilder builder = populateHearingCaseFields(hearingBooking);
caseDetails.getData().putAll(mapper.convertValue(builder, new TypeReference<>(){});